### PR TITLE
Fix devDep logic in `@lg-tools/validate`

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Updates the logic to check whether a dependency is used in a development file

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: tools/validate/src/dependencies/config.ts [hunk 1]
```diff
@@ -30,8 +30,15 @@ export const ignoreFilePatterns: Array<RegExp> = [
   /.*\/dist\/.*/,
 ];
 
-// these dependencies will be ignored when listed in a package.json
-export const ignoreMatches = [
+/** If a dependency is flagged as bing imported by one of these files, ignore it */
```
Comment (bruugey) [inline tools/validate/src/dependencies/config.ts:33]: nit: being

File: tools/validate/src/dependencies/config.ts [hunk 2]
```diff
@@ -30,8 +30,15 @@ export const ignoreFilePatterns: Array<RegExp> = [
...
+export const ignoreFilePatterns: Array<RegExp> = [
+  /.*package.json?/,
+  /.*README.md/,
+  /.*CHANGELOG.md/,
+];
+
+/** these dependencies will be ignored when listed in a package.json */
```
Comment (bruugey) [inline tools/validate/src/dependencies/config.ts:40]: how come?

Comment (TheSonOfThomp) [reply tools/validate/src/dependencies/config.ts:40]: These are globally available dev dependencies. We don't want every package flagged for not having React or Storybook or Jest explicitly declared in its package.json

Comment (bruugey) [reply tools/validate/src/dependencies/config.ts:40]: Leave that in the comment :)

**Review by bruugey — COMMENTED** [review]
how'd we test this? how can i?

**Review by bruugey — APPROVED** [review]

**Pull request merged**